### ISSUE TITLE
Number of classes: Real-CUR dataset

### ISSUE THREAD

**Issue by rnucuta** [YeolJ00/Personalized-Aesthetics#5]
The real-cur dataset has 5 aesthetic labels (1-5). Why does the train_piaa.py script use MultiModalIAA (10 aesthetic classes) instead of MultiModalPIAA  (5 aesthetic classes), but trains on the real-cur dataset? Are the provided pre-trained models only task vector tunable for the 10 class model architecture?

It is noted in the paper that "for tasks with score ranges from 1 to 5, we adjust the 10-dimensional template vector to include values from 0.5 to 5 in 0.5 increments." I don't think this is implemented in the code however. When I run the train script for Real-CUR, it seems the output values are in scale of 1-10, giving me a spearman correlation not matching the results of the paper (ran on worker idx 5).

Thank you!

**Comment by rnucuta** [comment]
I tried to scale the aesthetic scores of REAL-CUR (1-5) by 2 just by multiplying the score by 2 in the dataloader. This actually gave worse results even though I have confirmed that the model trains on a scale of 1-10. When training on worker_idx 5 (album 6) of the Real-CUR dataset, with default settings, i get plcc: 0.4854912153833293, srcc: 0.4559629074926795. When scaling the scores to be 1-10, i get plcc: 0.4203990502062612, srcc: 0.3877529008662672. This is with ViT-B/16. Please advise.

**Comment by YeolJ00** [comment]
Hi @rnucuta,

It looks like you're doing everything correctly. In theory, scaling the scores shouldn't affect final performance, since the loss is driven by preference rankings rather than absolute values. So it's a bit surprising to see a performance gap there. I would suggest double checking the codes there. 

Regarding your question (0.45 SRCC is not what's reported), this is likely due to variance in individual-specific performance. If you take a look at page 22 of [our arXiv paper](https://arxiv.org/pdf/2407.07176), you’ll notice that certain workers, like worker 5, are consistently harder to model, showing lower-than-average performance. That might explain the discrepancy you're seeing.

If you're aiming to further improve results, I'd recommend trying early stopping or tuning the temperature used for alpha initialization. In my experience, those were the most impactful parameters.

**Comment by rnucuta** [comment]
Thank you for your reply! Yes, I noticed results reflect the paper when using scales of 1-5 on Real-CUR when running other workers. It is quite odd that this changes with just editing one line however, even though rank loss doesn't take into account magnitude, just compares rank between two random samples. 

For reference, all i changed was one line: line 113 in `dataset.py` in `REALCURDataset`:
`data['MOS'] = float(meta['MOS'])` to `data['MOS'] = float(meta['MOS']) * 2`

I also noticed in the code cosine annealing is not implemented/min_lr is not used. Does this improve results significantly if implemented?

**Comment by YeolJ00** [comment]
Hi! Great to hear that it wasn't a serious issue. 

Cosine annealing can be a good choice if you have a good idea on the number of steps you need for a certain individual. But from my experience, it didn't give a huge boost in performance.
